Project-Aurora/Project-Aurora/Profiles/Discord/GSI/AuroraGSI.plugin.js

```diff
@@ -0,0 +1,248 @@
+		shell.Exec("explorer " + pathPlugins);
+		shell.Popup("I'm installed!", 0, "Successfully installed", 0x40);
+	}
+	WScript.Quit();
+
+@else@*/
+
+class AuroraGSI {
+	
+    getName() { return "AuroraGSI"; }
+    getDescription() { return "Sends information to Aurora about users connecting to/disconnecting from, mute/deafen status"; }
+    getVersion() { return "1.1"; }
+	getAuthor() { return "Popato & DrMeteor"; }
+	getChanges() {
+		return {
+            "1.0.0" : 
+            `
+                Initial version.
+            `,
+            "1.0.1" :
+            `
+                Added conditions for only reacting to local user.
+            `,
+            "1.0.2" :
+            `
+                Removed isBeingCalled.
+				Removed redundant loop.
+            `,
+            "1.0.3" :
+            `
+                Updated the CDN for the library.
+            `,
+            "1.1" :
+            `
+                Made the state only be sent if it changed.
+            `
+		};
+    }
+    
+    constructor(){
+        this.json = {
+            "provider": {
+                "name": "discord",
+                "appid": -1
+            },
+            "user":{
+                "id": -1,
+                "status": "undefined",
+                "self_mute": false,
+                "self_deafen" : false,
+                "mentions": false,
+                "unread_messages": false,
+                "being_called": false
+            },
+            "guild": {
+                "id": -1,
+                "name": "",
+            },
+            "text": {
+                "id": -1,
+                "type": -1,
+                "name": "",
+            },
+            "voice": {
+                "id": -1,
+                "type": -1,
+                "name": "",      
+            }
+        }
+        this.lastJson;
+    }
+
+    load() {}//legacy
+
+    start() {
+        let libLoadedEvent = () => {
+            try{ this.onLibLoaded(); }
+            catch(err) { console.error(this.getName(), "fatal error, plugin could not be started!", err); }
+        };
+
+		let lib = document.getElementById("NeatoBurritoLibrary");
+		if(lib == undefined) {
+			lib = document.createElement("script");
+			lib.setAttribute("id", "NeatoBurritoLibrary");
+			lib.setAttribute("type", "text/javascript");
+			lib.setAttribute("src", "https://raw.githack.com/Popat0/Discord-GSI/master/NeatoBurritoLibrary.js");
+			document.head.appendChild(lib);
+		}
+        if(typeof window.Metalloriff !== "undefined") libLoadedEvent();
+        else lib.addEventListener("load", libLoadedEvent);
+    }
+    
+    stop() {
+        clearInterval(this.updatetimer);
+		//this.unpatch();
+        this.ready = false;
+    }
+    
+	onLibLoaded() {
+        
+        this.settings = NeatoLib.Settings.load(this, this.defaultSettings);
+		NeatoLib.Updates.check(this, "https://raw.githubusercontent.com/Popat0/Discord-GSI/master/AuroraGSI.plugin.js");
```

Comment:
Why would you want another update check in the js code? Either use the pluginrepo from BD or do the update check here in C#

Reply:
You do realize not everyone uses PluginRepo, right? As for Aurora it only is supposed to install the plugin for a first run, not check if it's up to date.